File: server/src/main/java/me/ajfleming/tworoomsio/listeners/CardRequestListeners.java
package me.ajfleming.tworoomsio.listeners;

import com.corundumstudio.socketio.SocketIOClient;
import com.corundumstudio.socketio.annotation.OnEvent;
import lombok.RequiredArgsConstructor;
import me.ajfleming.tworoomsio.controller.UserActionController;
import me.ajfleming.tworoomsio.service.sharing.CardShareRequest;
import me.ajfleming.tworoomsio.socket.action.ShareDecisionRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class CardRequestListeners extends RequestListener {

  private final UserActionController userActionController;

  @OnEvent("REQUEST_SHARE")
  public void onRequestShare(SocketIOClient client, CardShareRequest request) {
    userActionController.requestShare(client, request);
  }

  @OnEvent("ACCEPT_SHARE")
  public void onAcceptShare(SocketIOClient client, ShareDecisionRequest request) {
    userActionController.acceptShare(client, request.getGameId(), request.getRequestId());
  }

  @OnEvent("REJECT_SHARE")
  public void onRejectShare(SocketIOClient client, ShareDecisionRequest request) {
    userActionController.rejectShare(client, request.getGameId(), request.getRequestId());
  }

  @OnEvent("PRIVATE_REVEAL")
  public void onPrivateReveal(SocketIOClient client, CardShareRequest request) {
    userActionController.privateReveal(client, request);
  }
}

File: kotlin_server/src/main/kotlin/me/ajfleming/tworoomsio/listeners/CardRequestListeners.kt
package me.ajfleming.tworoomsio.listeners

import com.corundumstudio.socketio.SocketIOClient
import com.corundumstudio.socketio.annotation.OnEvent
import me.ajfleming.tworoomsio.controller.PlayerActionController
import me.ajfleming.tworoomsio.socket.event.sharing.CardShareRequest
import me.ajfleming.tworoomsio.socket.event.sharing.ShareDecisionRequest
import org.springframework.stereotype.Component

@Component
class CardRequestListeners(
    val playerActionController : PlayerActionController
) : RequestListener() {

    @OnEvent("REQUEST_SHARE")
    fun onRequestShare(client: SocketIOClient, request: CardShareRequest) {
        playerActionController.requestShare(client, request);
    }

    @OnEvent("ACCEPT_SHARE")
    fun onAcceptShare(client: SocketIOClient, request: ShareDecisionRequest) {
        playerActionController.acceptShare(client, request.gameId, request.id)
    }

    @OnEvent("REJECT_SHARE")
    fun onRejectShare(client: SocketIOClient, request: ShareDecisionRequest) {
        playerActionController.rejectShare(client, request.gameId, request.id)
    }

    @OnEvent("PRIVATE_REVEAL")
    fun onPrivateReveal(client: SocketIOClient, request: CardShareRequest) {
        playerActionController.privateReveal(client, request)
    }
}